### PR TITLE
Universal Link redirect acceptance

### PR DESCRIPTION
**Pull request by unpluggedk**
This would be used in MSAL, 
PR regarding broker will come in a separate PR

Being able to handle universal link response from broker and system webview.

### CODE REVIEW

File: IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m [hunk 1]
```diff
@@ -199,5 +198,23 @@ - (void)notifyEndWebAuthWithURL:(NSURL *)url
     }
 }
 
+- (BOOL)handleURLResponse:(NSURL *)url
+{
+    [[MSIDTelemetry sharedInstance] stopEvent:_telemetryRequestId event:_telemetryEvent];
+    
+    if (@available(iOS 12.0, *))
```
Comment (oldalton) [inline IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m]: Double check that this is not going to call the cancellation callback before we can execute with notifyEndWebAuthWithURL.

File: IdentityCore/src/webview/systemWebview/ios/MSIDSystemWebviewController.m [hunk 2]
```diff
@@ -56,7 +57,8 @@ - (instancetype)initWithStartURL:(NSURL *)startURL
         return nil;
     }
     
-    if (!callbackURLScheme)
+    NSURL *redirectURL = [NSURL URLWithString:redirectURI];
+    if (!redirectURL)
```
Comment (oldalton) [inline IdentityCore/src/webview/systemWebview/ios/MSIDSystemWebviewController.m]: nit: can you also check for scheme presence here as technically scheme is nullable on NSURL

File: IdentityCore/tests/mocks/MSIDTestWebviewInteractingViewController.m [hunk 3]
```diff
@@ -55,17 +55,17 @@ - (void)cancel
...
+    if (self.actAsSafariViewController || self.actAsAuthenticationSession)
     {
         return (aClass == MSIDSystemWebviewController.class);
     }
 #endif
     return NO;
 }
 
-- (BOOL)handleURLResponseForSafariViewController:(NSURL *)url
+- (BOOL)handleURLResponse:(NSURL *)url;
```
Comment (rohitnarula7176) [inline IdentityCore/tests/mocks/MSIDTestWebviewInteractingViewController.m]: nit: Remove semicolon.

File: IdentityCore/tests/MSIDSystemWebviewControllerTests.m [hunk 4]
```diff
@@ -25,6 +25,16 @@
 
 #import <XCTest/XCTest.h>
 #import "MSIDSystemWebviewController.h"
+#import "MSIDURLResponseHandling.h"
+
+@interface MSIDTestSession: NSObject<MSIDURLResponseHandling>
+@end
+
+@implementation MSIDTestSession: NSObject
+- (BOOL)handleURLResponse:(NSURL *)url {
```
Comment (antrix1989) [inline IdentityCore/tests/MSIDSystemWebviewControllerTests.m]: nit: new line

File: IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m (hunk 1 above)
Comment (unpluggedk) [reply IdentityCore/src/webview/systemWebview/ios/MSIDAuthenticationSession.m]: checked, authentication session has no delegates in current context. 
thus this is fine

File: IdentityCore/tests/MSIDSystemWebviewControllerTests.m (hunk 4 above)
Comment (unpluggedk) [reply IdentityCore/tests/MSIDSystemWebviewControllerTests.m]: done

**Pull request merged**